modules/extension/wmts/src/main/java/org/geotools/ows/wmts/map/WMTSCoverageReader.java

```diff
@@ -366,10 +383,10 @@ protected void renderTile(Tile tile, Graphics2D g2d, double[] points) {
         }
         g2d.drawImage(
                 img,
-                (int) points[0],
-                (int) points[1],
-                (int) Math.ceil(points[2] - points[0]),
-                (int) Math.ceil(points[3] - points[1]),
+                (int) Math.round(points[0]),
```

Comment:
Maybe I'm not getting the point but using Math.round looks the better way to me since it provides the closest value to the double one. Also by doing some live test with both the solutions, this is the way I got the clearest output.

Reply:
Been trough it too, there are a number of positions very close to the unit, using floor and ceil causes 1-2 pixels to be added and quality suffers quite visibly due to a rescale being necessary after that. @taba90 I'd still make sure that the box hits at least one pixel in both directions though.

Reply:
@aaime fixed